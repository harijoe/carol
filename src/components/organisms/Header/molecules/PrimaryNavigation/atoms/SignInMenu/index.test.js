import React from 'react'
import { shallow } from 'enzyme'
import MenuSignIn from '.'

const wrap = (props = {}) => shallow(<MenuSignIn {...props} />)

describe('user is not connect', () => {
  it('should render a form', () => {
    const wrapper = wrap({ isAuthenticated: false })

    expect(wrapper.find('form')).toHaveLength(1)
  })
})

describe('user is connect with grantType facebook', () => {
  it('should render MenuAccount with no form', () => {
    const wrapper = wrap({ isAuthenticated: true })

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('form')).toHaveLength(0)
  })
})
