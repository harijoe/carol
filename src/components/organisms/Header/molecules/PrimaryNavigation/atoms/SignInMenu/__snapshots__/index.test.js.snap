exports[`user is connect with grantType facebook should render MenuAccount with no form 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Mock
    label="user.my_account">
    <Mock />
</Mock>,
  "nodes": Array [
    <Mock
      label="user.my_account">
      <Mock />
</Mock>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SignInMenu
        isAuthenticated={true} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isAuthenticated": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Mock
          label="user.my_account">
          <Mock />
</Mock>,
        "_debugID": 4,
        "_renderedOutput": <Mock
          label="user.my_account">
          <Mock />
</Mock>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignInMenu
    isAuthenticated={true} />,
}
`;
